rskj-core/src/main/java/org/ethereum/core/BlockHeader.java

```diff
@@ -366,7 +370,11 @@ public byte[] getEncoded(boolean withMergedMiningFields, boolean withMerkleProof
             fieldToEncodeList.add(RLP.encodeElement(this.ummRoot));
         }
 
-        this.addExtraFieldsToEncoded(useExtensionEncoding, fieldToEncodeList);
+        if (this.getVersion() > 0) {
+            fieldToEncodeList.add(RLP.encodeByte(this.getVersion()));
```

Comment:
Why the version field is encoded always ? It should only be encoded in uncompressed format. In compressed format, the version would be in the logsBloom field. Right ?

Reply:
I was standing on that while implementing and I didn’t lime how it looked 🙂 I think this has the same result of what we discussed. Now logs bloom field is 32 bytes long for version > 0  in that case you should read version from RLP element at the end

Reply:
Ok so we are going to keep the RLP size of the _compressed encoding V1_ equal to the RLP size of the _encoding V0_ (prior hard fork).

As `txEdges` and `version` are added on V1 and that `logsBloom` field is replaced by `RLP(version, extensionHash)`, I'm adding this test


```java
@Test
    public void compressedEncodingV1HasSameRLPSizeAsFullEncodedV0WithoutEdges () {
        // this test is added to assert that the rlp header size does not change
        // in the hard fork, assuming both RSKIP 351 and RSKIP 144 are activated
        // together
        byte[] logsBloom = new byte[Bloom.BLOOM_BYTES];
        logsBloom[0] = 0x01;
        logsBloom[1] = 0x02;
        logsBloom[2] = 0x03;
        logsBloom[3] = 0x04;

        BlockHeaderV0 headerV0 = (BlockHeaderV0) createBlockHeaderWithVersion((byte) 0x0);
        headerV0.setLogsBloom(logsBloom);
        headerV0.setTxExecutionSublistsEdges(null);

        BlockHeaderV1 headerV1 = (BlockHeaderV1) createBlockHeaderWithVersion((byte) 0x1);
        headerV1.setLogsBloom(logsBloom);
        headerV1.setTxExecutionSublistsEdges(null);

        RLPList compressedEncodingV1 = RLP.decodeList(headerV1.getEncodedCompressed());
        RLPList fullEncodedV0 = RLP.decodeList(headerV0.getFullEncoded());

        Assertions.assertEquals(fullEncodedV0.size(), compressedEncodingV1.size());
    }
```